internal/daemon/daemon_test.go

```
@@ -600,3 +603,159 @@ func TestListMessagesWithoutForReturnsEventLog(t *testing.T) {
+	go func() { serveErr <- d.Start(ctx) }()
+
+	// Wait until the daemon is accepting connections so we know Start is past
+	// runtime.Up and into Serve. Drain serveErr to surface any fast-fail.
+	ready := false
+	for deadline := time.Now().Add(2 * time.Second); time.Now().Before(deadline); {
+		select {
+		case err := <-serveErr:
+			t.Fatalf("Start returned early: %v", err)
+		default:
+		}
+		c, dialErr := net.Dial("unix", socketPath)
```

golangci-lint (2.11.4)

[error] net.Dial must not be called. use (*net.Dialer).DialContext

(noctx)